File: ApiBookStore/DTO/OrderDto.cs
using ApiBookStore.Entities;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace ApiBookStore.DTO
{
    public class OrderDto
    {
        public int OrderId { get; set; }
        public int? UserId { get; set; }
        public int? ShippingAddressId { get; set; }
        public DateOnly OrderDate { get; set; }
        public int Status { get; set; }
        public List<OrderItemDto> Books { get; set; } = new List<OrderItemDto>();
    }
}

File: BookShop/src/app/Dto/order-dto.ts
import { iOrderItemDto } from "./order-item-dto";
import { iShippingAddressDto } from "./shipping-address-dto";

export interface iOrderDto {
    orderId: number;
    userId: number | null;
    shippingAddressId: number | null;
    orderDate: Date;
    status: number;
    shippingAddress: iShippingAddressDto | null;
    books: iOrderItemDto[];
}
